src/main/java/com/joyent/manta/cosbench/MantaStorage.java

```diff
@@ -241,9 +274,45 @@ public void createObject(final String container,
         }
     }
 
+    /**
+     * Helper method for parsing out the streams and uploading in the multi-part way.
+     *
+     * @param data - Data stream.
+     * @param path - The path that we are going to put the object into.
+     * @param multipartManager - This will be EncryptedServerSideMultipartManager or ServerSideMultipartManager.
+     */
+    @SuppressWarnings("unchecked")
+    private void multipartUpload(final InputStream data, final String path,
+            @SuppressWarnings("rawtypes") final MantaMultipartManager multipartManager) {
+        MantaMultipartUpload upload = null;
+        try {
+            upload = multipartManager.initiateUpload(path);
+            logger.info("Split Size : {} ", splitSize);
+            int splits = objectSize / splitSize;
+            // Just a check to see if we should do one more split.
+            if ((objectSize % splitSize) != 0) {
+                splits++;
+            }
+            logger.info("Splitting file int {}", splitSize);
+            LinkedList<MantaMultipartUploadPart> parts = new LinkedList<MantaMultipartUploadPart>();
+            for (int i = 0; i < splits; i++) {
+                try (BoundedInputStream bis = new BoundedInputStream(data, objectSize)) {
+                    parts.add(multipartManager.uploadPart(upload, i + 1, bis));
+                    logger.debug("grabbed the next {} bytes", splitSize);
+                } catch (Exception e) {
+                    logger.error("Error in putting together the MPU {}", e.getMessage());
+                }
```

Comment:
It would probably make the results and logs easier to debug if this exception were rethrown here. Wouldn't continuing the for loop and completing the MPU lead to inaccurate results if there is an error during one of the parts?